plugins/gatsby-source-data/main.js

```diff
@@ -140,31 +139,40 @@ const getPlaylistsFromYoutube = async (orgData, apiKey) => {
           const playlistObj = {
             id: playlist.id,
             title: playlist.snippet.title,
-            organisation: playlist.snippet.channelTitle,
-            donationMethod: "Paynow to UEN " + org.paynowUen,
-            language: tags.language.join(','),
-            tags: tags.category.join(','),
-            platform: "YouTube",
-            pageUrl: org.youtubeUrl || "",
+            channelTitle: playlist.snippet.channelTitle,
+            publishedAt: playlist.snippet.publishedAt,
+            organisationName: org.organisationName,
+            donationUrl: org.donationUrl,
+            language: tags.language,
+            tags: tags.category,
             thumbnailUrl: playlist.snippet.thumbnails.medium.url, // Encountered error: thumbnails key (i.e. thumbnails.standard) may not exist.
-          };
+            youtubeMeta: playlist,
+          };        
 
           const playlistVideos = await yt.getPlaylistVideos(playlist.id);
-          const videosArr = playlistVideos.map(video => {
-            return {
-              id: video.snippet.resourceId.videoId,
-              playlistId: video.snippet.playlistId,
-              title: video.snippet.title,
-              asatizah: "<Asatizah name>", // TODO: Parse from description/title?
-              language: tags.language[0],  // TODO: Frontend needs to take an array instead.
-              addedOn: video.snippet.publishedAt,
-              videoUrl:
-                "https://www.youtube.com/embed/" + video.snippet.resourceId.videoId,
-            };
+          const videosArr = []
```

Comment:
1. Use functional methods for better readability
2. Never append the object type to the name of the variable (e.g `videosArr`, call it `videos` instead)
```
videos = playlistVideos
  .filter(video => video.snippet.title !== 'Private video')
  .map(video => ({
    id: video.snippet.resourceId.videoId,
    title: video.snippet.title,
    description: video.snippet.description,
    publishedAt: video.snippet.publishedAt,
    thumbnailUrl: video.snippet.thumbnails.medium.url,
    youtubeMeta: video,
  }))

  latestDate = videos.reduce((date, video) =>
    video.snippet.publishedAt > date ? video.snippet.publishedAt : date, new Date(0))
```